tools/release/lp_build_monitor_recipe.py

```diff
@@ -0,0 +1,157 @@
+import sys
+import time
+import argparse
+import textwrap
+import itertools
+
+from lazr.restfulclient.errors import BadRequest
+
+from utils import get_build_recipe, get_date_utc_now
+
+"""
+This script builds a recipe (for the platforms requested on LP),
+monitoring the build process and automatically retrying any build
+that fails. If Launchpad doesn't allow to retry a build, the script
+will keep on monitoring the others, and then exit with a non-0 return
+value.
+"""
+
+# delay between updates requested to LP
+LP_POLLING_DELAY = 60
+
+
+def start_all_build(build_recipe):
+    for series in build_recipe.distroseries:
+        try:
+            build_recipe.requestBuild(
+                pocket="Release",
+                distroseries=series,
+                archive=build_recipe.daily_build_archive_link,
+            )
+        except BadRequest:
+            print("An identical build of this recipe is already pending")
+
+
+def wait_every_build_started(build_recipe):
+    pending_builds = build_recipe.getPendingBuildInfo()
+    while pending_builds:
```

Comment:
This along with trying to resolve "unrecoverable builds" is a good future expansion direction. If a build is stuck in the pending queue, I think there is not much we can do here. We could fail the workflow if one build is stuck for >timeout in the queue, but I don't expect this to happen. The reality is that this will wait for Xhours, till all builds are ready to start and the next step will find most of the others that were not stuck (non risc/armhf builds for example) completed. Are we loosing some time where we could be retrying builds? yes. I hope it is not a huge issue, else we will tackle in a future PR. wdyt?

Reply:
not an issue, just voicing my concerns